### PR TITLE
fix firefox drag and drop abandon

### PR DESCRIPTION
**Pull request by chloeleichen**
Fixed the issue that you cannot abandon drag and drop in firefox (see https://github.com/TerriaJS/terriajs/issues/3036).

One of the changes is removing the `screenX` and `screenY` check in `handleDragLeave` function in `DragDropFile.jsx`. I don't see why that line is necessary, and it seems that firefox does not set those values to 0 when the drag is finished. Please let me know if that check is there for other reasons.

### CODE REVIEW

File: lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx [hunk 1]
```diff
@@ -57,16 +57,26 @@ const StandardUserInterface = createReactClass({
...
         const that = this;
-        this.dragOverListener = e => {
+
+        this.dragEnterListener = e => {
             if (!e.dataTransfer.types || !arrayContains(e.dataTransfer.types, 'Files')) {
                 return;
             }
             e.preventDefault();
             e.stopPropagation();
-            e.dataTransfer.dropEffect = 'copy';
```
Comment (kring) [inline lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx]: Are you sure this isn't needed for IE or something?

File: lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx [hunk 2]
```diff
@@ -57,16 +57,26 @@ const StandardUserInterface = createReactClass({
...
+            console.log("drag enter")
```
Comment (kring) [inline lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx]: Please remove this.

File: lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx (hunk 1 above)
Comment (kring) [reply lib/ReactViews/StandardUserInterface/StandardUserInterface.jsx]: Oh I see you just moved it, nevermind.

**Comment by kring** [comment]
This isn't working very well in Chrome on Windows for me. It flickers as I move the mouse around. See the gif below. I suspect because a "dragleave" event is being generated for individual element, not just for the whole window.
drag-drop

**Comment by chloeleichen** [comment]
Found a better fix! @kring ready for another look! 💃 💃 💃

**Comment by kring** [comment]
Thanks @chloeleichen!

**Pull request merged**